### DISCUSSION TITLE
student changed the github username

### DISCUSSION THREAD

**Discussion by arademaker** [community/community#136501] [GitHub Education]
### Hi everyone,

One of my students changed his username on GitHub. In the classroom, I can unlink the roster entry, but the new GitHub username does not appear when I try to link the roster entry to a GitHub account again.

Any idea?

**Comment by GabyCT** [comment]
According to the Github documentation
[Repository references](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-user-account-settings/changing-your-github-username#repository-references)
After you change your username, GitHub will automatically redirect references to your repositories.

Web links to your existing repositories will continue to work. This can take a few minutes to complete after you make the change.
Command line pushes from your local repository clones to the old remote tracking URLs will continue to work.
If the new owner of your old username creates a repository with the same name as your repository, that will override the redirect entry and your redirect will stop working. Because of this possibility, we recommend you update all existing remote repository URLs after changing your username.

[Accounts logged in on GitHub Mobile](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-user-account-settings/changing-your-github-username#accounts-logged-in-on-github-mobile)
Accounts logged in on the GitHub Mobile app may continue to display your original username until you log out. To ensure your updated username is displayed, we recommend you sign out and back in to your account on each mobile device.

For more information https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-user-account-settings/changing-your-github-username

**Comment by arademaker** [comment]
Yes, but my question is about the interaction between GitHub and the GitHub Classroom... I am afraid he was still linked to the old GitHub username on the roster. I unliked him, and I hope he will be relinked from the new account in the next assignment.

**Comment by davevad93** [comment]
Hi @arademaker ,

According to a [comment](https://github.com/github-education-resources/classroom/issues/2508#issuecomment-583033680) found on [issue #2508](https://github.com/github-education-resources/classroom/issues/2508) in the [classroom](https://github.com/github-education-resources/classroom) repository, one way to fix this is to change back to the previous username linked to the classroom.

**Comment by arademaker** [comment]
I was able to fix removing the student from the roster and in the next assignment, he appeared again with the new login as an "outside collaborator"  but with the correctly new github userid linked to the student in the classroom.
